sdk/core/core/src/az_precondition.c

```diff
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: MIT
 
 #include <az_platform_internal.h>
-#include <az_precondition.h>
+#include <az_precondition_internal.h>
```

Comment:
I think this should not be here and az_precondition.h should be here to make sure the signatures match.

Reply:
explained on call